catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/converter/DorisExceptionConverter.java

```diff
@@ -6,15 +6,70 @@
 
 import com.datastrato.gravitino.catalog.jdbc.converter.JdbcExceptionConverter;
 import com.datastrato.gravitino.exceptions.GravitinoRuntimeException;
+import com.datastrato.gravitino.exceptions.NoSuchColumnException;
+import com.datastrato.gravitino.exceptions.NoSuchSchemaException;
+import com.datastrato.gravitino.exceptions.NoSuchTableException;
+import com.datastrato.gravitino.exceptions.SchemaAlreadyExistsException;
+import com.datastrato.gravitino.exceptions.TableAlreadyExistsException;
+import com.datastrato.gravitino.exceptions.UnauthorizedException;
+import com.google.common.annotations.VisibleForTesting;
 import java.sql.SQLException;
+import java.util.regex.Pattern;
 
 /** Exception converter to Gravitino exception for Doris. */
 public class DorisExceptionConverter extends JdbcExceptionConverter {
+
+  // see: https://doris.apache.org/docs/admin-manual/maint-monitor/doris-error-code/
+  @VisibleForTesting static final int CODE_DATABASE_EXISTS = 1007;
+
+  static final int CODE_TABLE_EXISTS = 1050;
+  static final int CODE_NO_SUCH_SCHEMA = 1049;
+  static final int CODE_DATABASE_NOT_EXISTS = 1008;
+  static final int CODE_NO_SUCH_TABLE = 1051;
+  static final int CODE_UNAUTHORIZED = 1045;
+  static final int CODE_NO_SUCH_COLUMN = 1054;
+  static final int CODE_OTHER = 1105;
+
+  private static final String DATABASE_ALREADY_EXISTS_PATTERN_STRING =
+      ".*detailMessage = Can't create database '.*'; database exists";
```

Comment:
is `.*detailMessage` the standard error message format for Doris?

Reply:
Doris does not use the standard SQLException ErrorCode, so we need to parse detailMessage.